packages/blade/src/components/Menu/_decisions/decisions.md

```diff
@@ -0,0 +1,308 @@
+  <Button>Open Menu</Button>
+  <MenuOverlay>
+    <MenuHeader title="" />
+    <MenuBody>
+      {/* All of this is a slot */}
+      <Box display="flex" paddingY="spacing.4" gap="spacing.2">
+        <Avatar name="Saurabh Daware" />
+        <Text>Saurabh Daware</Text>
+      </Box>
+      <MenuDivider />
+      <MenuItem title="Undo" onClick={() => console.log('Undo')} />
+      <MenuItem title="Redo" isDisabled />
+      <MenuItem title="Cut" />
+      {/* Sub Menu */}
+      <Menu>
+        <MenuItem title="Copy" />
+        <MenuOverlay>
+          <MenuItem title="Text" />
+          <MenuItem title="Video" />
+          <MenuItem title="Audio" />
+        </MenuOverlay>
+      </Menu>
+      <Menu>
+        <MenuItem title="Share" />
+        <MenuOverlay>
+          <MenuItem title="Text" />
+          <MenuItem title="Video" />
+          <MenuItem title="Audio" />
+        </MenuOverlay>
+      </Menu>
+    </MenuBody>
+    <MenuFooter>
+      <Button>Apply</Button>
+    </MenuFooter>
+  </MenuOverlay>
+</Menu>
+```
+
+## Props
+
+### Menu
+
+```ts
+type MenuProps = {
+  /**
+   * First children is trigger and second children is MenuOverlay
+   **/
+  children: [React.ReactElement, React.ReactElement];
+
+  /**
+   * Open controlled state
+   */
+  isOpen?: boolean;
+
+  /**
+   * On Menu open change callback
+   */
+  onOpenChange?: (isOpen: boolean) => void;
+
+  /**
+   * Should menu open on click or hover
+   *
+   * @default 'click'
+   */
+  openInteraction?: 'hover' | 'click';
+};
+```
+
+### MenuOverlay
+
+This overlay will be flexible overlay which can have any component. Static / Non-Static / MenuItem / etc
+
+```ts
+type MenuOverlay = {
+  /**
+   * A slot inside Menu's overlay.
+   *
+   * Supports MenuItem or any other component
+   */
+  children: React.ReactElement[];
+};
+```
+
+### MenuItem
+
+```ts
+type MenuItemProps = {
+  /**
+   * title of item
+   */
+  title: string;
+
+  /**
+   * Description text for the item
+   */
+  description?: string;
+
+  /**
+   * Click handler for MenuItem
+   *
+   * Absense of this prop and href will turn the item into non-interactive item
+   */
+  onClick?: (event: React.MouseEvent) => void;
+
+  /**
+   * Link to open when item is clicked.
+   *
+   * Absense of this prop and onClick will turn the item into non-interactive item
+   */
+  href?: string;
+
+  /**
+   * HTML target of the link
+   */
+  target?: string;
+
+  /**
+   * Item that goes on left-side of item.
+   *
+   * Will be overriden in multiselect
+   */
+  leading?: React.ReactNode;
+
+  /**
+   * Item that goes on right-side of item.
+   */
+  trailing?: React.ReactNode;
+
+  /**
+   * Item that goes immediately next to the title.
+   */
+  titleSuffix?: React.ReactElement;
+
+  /**
+   * disabled state of item
+   */
+  isDisabled?: boolean;
+
+  /**
+   * Color of item. set to negative for dangerous actions like Delete, Remove, etc
+   */
+  color?: Extract<FeedbackColors, 'negative'>;
```

Comment:
This type doesn't make sense? Extracting negative from FeedbackColors means color will always be negative?